setup.py

```diff
@@ -162,23 +163,25 @@ def _collect_requirements(requires):
     "jupyter~=1.0",
     "lxml~=4.6",
     "matplotlib>=3.0.3, <3.4; python_version < '3.10'",  # 3.4.0 breaks holoviews
-    "matplotlib>=3.5, <3.6; python_version == '3.10'",
+    "matplotlib>=3.5, <3.6; python_version >= '3.10'",
     "memory_profiler>=0.50.0, <1.0",
     "moto==1.3.7; python_version < '3.10'",
-    "moto==3.0.4; python_version == '3.10'",
+    "moto==4.1.12; python_version >= '3.10'",
     "networkx~=2.4",
     "opencv-python~=4.5.5.64",
     "openpyxl>=3.0.3, <4.0",
-    "pandas-gbq>=0.12.0, <0.18.0",
+    "pandas-gbq>=0.12.0, <0.18.0; python_version < '3.11'",
+    "pandas-gbq>=0.18.0; python_version >= '3.11'",
     "pandas~=1.3  # 1.3 for read_xml/to_xml",
     "Pillow~=9.0",
     "plotly>=4.8.0, <6.0",
     "pre-commit>=2.9.2, <3.0",  # The hook `mypy` requires pre-commit version 2.9.2.
-    "psutil~=5.8",
-    "pyarrow>=6.0",
+    "pyarrow>=1.0; python_version <= '3.10'",
+    "pyarrow>=7.0; python_version > '3.10'",
```

Comment:
I forgot what's the reason that requires pyarrow >=7.0, can we add a comment here?

In addition, a styling problem. We use two pattern inconsitently, will be good to keep one.
```
xxx <= "3.10"
xxx > "3.10"
```

or 
```
xxx < "3.11"
xxx >= "3.11"
```

Actually, this is a bug if I didn't mess up.  <=3.10 mean anything before 3.10 which doesn't include 3.10.x. I think it's better to stick with pattern 2 here.

Reply:
I think it was to avoid the numpy build errors, pinning pyarrow >=7.0 seems to have resolved it.